### ISSUE TITLE
Usage of PWM

### ISSUE THREAD

**Issue by christoph2307** [DeusVermis/heating_control#3]
Hey,

I have built my own heating control based on your repo. Thanks for the great work.
However, the dimming range can be improved. Using PWM the resolution can be increased. In my case it was from 650-1024 which is about equal to your range from 80-100. At first it was not working but increasing the pwm frequency to 10 kHz, everything works fine.

**Comment by LordTompa** [comment]
Hi, could you explain your process in more detail? 
Did you use the same hardware as in the repo or did you make changes?
How did you configure the software to get 10KHz?
I am planning to rebuild the whole thing myself and am in the process of getting the parts together, so I could make some changes if you have any recommendations.

Thanks in advance

**Comment by christoph2307** [comment]
Hi, I built the network based on your schematic.
I used all the hardware including the ESP8266 and the Tasmota software.

You can configure a PWM frequency of up to 20 kHz in the config of the Tasmota firmware. The default is limited to 4 kHz which seems not to be enough for the Vaillant. It will just turn on and heat to the maximum temperature and ignores the set PWM values. I increased the PWM frequency because I read something about that in a forum (do not know where). And after increasing the PWM frequency to 10 kHz, it works and you have a huge range to set the temperature from.

I did have my problems flashing the firmware on the device in the first try, so be sure that your pins are connected correctly. That is the only recommendation I can give you.

**Comment by LordTompa** [comment]
Hi thanks for the reply,
Is it a standard setting to increase the pwm frequency in tasmota to this level or do I need to flash a custom firmware?
Thanks in advanced

**Comment by christoph2307** [comment]
In tasmota.h you need to change the constant
const uint16_t PWM_MAX 
from 4000 to 20000 and then compile the tasmota firmware. The resulting bin can be flashed via a serial flasher (before you need flash the lite firmware to make space for the bigger firmware). After that you can set the PWM frequency to your liking

**Comment by StefanLobbenmeier** [comment]
> I increased the PWM frequency because I read something about that in a forum (do not know where)

Might have been here: https://github.com/arendst/Tasmota/issues/9238 

> After that you can set the PWM frequency to your liking

Just to elaborate on that, this is the documentation on that: https://tasmota.github.io/docs/Commands/#commands-list
Command is entered in the tasmota website, click on "Command", enter `PWMFrequency 10000`
I do not know which command you used to change dimmer range from 0-100

**Comment by christoph2307** [comment]
The PWM frequency is set in the configuration files for Tasmota, Then you have to compile the firmware by yourself and flash the binary to the ESP32.
After this you can use the PWM command with values from 0-1024. 
On the web interface you set the PWM from 0-1024. I set mine using the http api: http://<IPofTasmota>/cm?cmnd=PWM2%20100

Hope this helps

**Comment by StefanLobbenmeier** [comment]
I see, so instead of setting the dimmer value from 0-100 you set the PWM value which is a different value and you completely ignore the “dimmer” Value, right?

**Comment by christoph2307** [comment]
Right, I set my GPIO16 to PWM 2 and then set the PWM value via http
